packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -0,0 +1,444 @@
+      }
+    });
+  }
+
+  void _handlePointerDown(PointerDownEvent event) {
+    _lastPointerLocation = event.position;
+    _inspectAt(event.position);
+  }
+
+  void _handlePointerMove(PointerMoveEvent event) {
+    _lastPointerLocation = event.position;
+    _inspectAt(event.position);
+  }
+
+  void _handlePointerCancel(PointerCancelEvent event) {
+    setState(() {
+      isSelectMode = false;
+      _selected = null;
+    });
+  }
+
+  void _handlePointerUp(PointerUpEvent event) {
+    _lastPointerLocation = event.position;
+
+    // If the pointer up position is on the edge of the window assume that it
+    // indicates the pointer is being dragged off the edge of the display not
+    // a regular mouse up near the edge of the display. If the pointer is being
+    // dragged off the edge of the display we do not want to select anything.
+    final Rect bounds =
+        (Offset.zero & (ui.window.physicalSize / ui.window.devicePixelRatio))
+            .deflate(_kOffScreenMargin);
+    if (!bounds.contains(event.position)) {
+      setState(() {
+        _selected = null;
+      });
+      return;
+    }
+    _inspectAt(event.position);
+  }
+
+  void _handleTap() {
+    if (!isSelectMode)
+      return;
+    if (_lastPointerLocation == null)
+      return;
+    setState(() {
+      _inspectAt(_lastPointerLocation);
+
+      if (_selected != null) {
+        // Notify debuggers to open an inspector on the object.
+        developer.inspect(_selected);
+      }
+      isSelectMode = false;
+    });
+  }
+
+  void _handleEnableSelect() {
+    setState(() {
+      isSelectMode = true;
+    });
+  }
+
+  // TODO(jacobr): This shouldn't be a static. We should get the pipeline owner
+  // from [context] somehow.
+  PipelineOwner get _pipelineOwner => WidgetsBinding.instance.pipelineOwner;
+
+  @override
+  Widget build(BuildContext context) {
+    return new CustomPaint(
+        foregroundPainter: new _WidgetInspectorPainter(
+            _pipelineOwner, _client.generation, _selected),
+        child: new Stack(children: <Widget>[
+          new GestureDetector(
+              onTap: _handleTap,
+              behavior: HitTestBehavior.opaque,
+              excludeFromSemantics: true,
+              child: new Listener(
+                  onPointerDown: _handlePointerDown,
+                  onPointerMove: _handlePointerMove,
+                  onPointerUp: _handlePointerUp,
+                  onPointerCancel: _handlePointerCancel,
+                  behavior: HitTestBehavior.opaque,
+                  child: isSelectMode
+                      ? new IgnorePointer(
+                          key: _ignorePointerKey,
+                          ignoringSemantics: false,
+                          child: widget.child)
+                      : widget.child)),
+          new Positioned(
+            left: _kInspectButtonMargin,
+            bottom: _kInspectButtonMargin,
+            child: isSelectMode
+                ? new Container()
```

Comment:
just exclude this entire subtree if you don't want it (by not including it in the stack child list).